app/models/solidus_afterpay/gateway.rb

```diff
@@ -122,8 +130,15 @@ def create_checkout(order, gateway_options)
 
     def find_payment(order_id:)
       ::Afterpay::API::Payment::Find.call(order_id: order_id).body
-    rescue ::Afterpay::BaseError
-      nil
+    rescue ::Afterpay::BaseError => e
+      ActiveMerchant::Billing::Response.new(false, e.message, {}, error_code: e.error_code)
```

Comment:
Are you sure that this change doesn't break this line https://github.com/nebulab/solidus_afterpay/blob/express-checkout/app/models/solidus_afterpay/payment_source.rb#L18?

Reply:
Yes, I checked it

Reply:
In that case, you need to return an `ActiveMerchant::Billig::Response` also when the API doesn't raise any error. Otherwise, you will have two different behaviors.